File: C++/greatest_sub_range_2.cpp
#include <iostream>

using namespace std;

int main()
{
    int n;
    cin >> n;
    int a[n];
    for (int i = 0; i < n; i++)
        cin >> a[i];

    long long int ans = -1000000000;
    for (int l = 0; l < n; l++)
    {
        long long int sum = 0;
        for (int r = l; r < n; r++)
        {
            sum += a[r];
            if (ans < sum)
                ans = sum;
        }
    }

    cout << ans << endl;
    return 0;
}

File: Python/greatest_sub_range_2.py
###first approach
import sys
num = int(input())
in_list = list(map(int, input().split(" ")))


ans = -(sys.maxsize - 1)
for left in range(len(in_list) + 1):
    sum = 0
    for right in range(left, len(in_list)):
        sum += in_list[right]

        if ans < sum:
            ans = sum
print(ans)

###second approach
n = int(input())
a = list(map(int, input().split()))
ans = -10 ** 9

for l in range(n):
    s = 0
    for r in range(l, n):
        s += a[r]
        if s > ans:
            ans = s

print(ans)
